core/tauri/src/api/notification.rs

```diff
@@ -72,6 +74,35 @@ impl Notification {
     self
   }
 
+  /// Sets the notification sound. Silent by default.
+  ///
+  /// ## Platform-specific
+  ///
+  /// Each OS has a different sound name so you will need to conditionally specify an appropriate sound
+  /// based on the OS in use, for a list of sounds see:
+  /// - **Linux**: can be one of the sounds listed in <https://0pointer.de/public/sound-naming-spec.html>
+  /// - **Windows**: can be one of the sounds listed in <https://learn.microsoft.com/en-us/uwp/schemas/tiles/toastschema/element-audio>
+  ///   but without the prefix, for example, if `ms-winsoundevent:Notification.Default` you would use `Default` and
+  ///   if `ms-winsoundevent:Notification.Looping.Alarm2`, you would use `Alarm2`.
+  ///   Windows 7 is not supported, if a sound is provided, it will play the default sound, otherwise it will be silent.
+  /// - **macOS**: you can specify the name of the sound you'd like to play when the notification is shown
+  /// or use `NSUserNotificationDefaultSoundName` to play the default sound.
```

Comment:
Will this actually play the default sound on macOS? if so I think we should just add a `.default_sound()` method

Reply:
I've pushed a change to use a sound enum instead, let me know what you think (mac-notification-sys did the same).

Reply:
and yeah it does play the default sound on macOS. The problem is the new mac-notification-sys methods aren't exposed by notify-rust, so we need to hardcode the values here.